### ISSUE TITLE
Build failure not recorded if failure during crate fetch step

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/docs.rs#797]
If you run

```console
> docker-compose run web build crate alcro 0.1.4
```

it will fail because of the lockfile (currently at least, this might be a bug). This is not recorded anywhere, so if it's built via the queue the version just silently disappears.

**Comment by jyn514** [comment]
Note that we still have a record of the crate: it's in the `queue` table with attempts set to 5. We just don't display that anywhere on the site - I think under the assumption this is an internal bug, but that's not always the case (#903).

**Comment by jyn514** [comment]
I think this can't be fixed without changes to rustwide. In particular, I don't think there's anything docs.rs can do without at least being able to parse the Cargo manifest, and we can't parse the manifest until we have access to the source directory: https://github.com/rust-lang/docs.rs/blob/ba9364b935e2ef1c9b925f4295d42fd03b9587dd/src/docbuilder/rustwide_builder.rs#L310
And rustwide generates its own errors before ever running that closure :/ the error from `alcro` looks like
```
   5: rustwide::prepare::Prepare::fetch_deps::{{closure}}
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/prepare.rs:169
   6: core::result::Result<T,E>::map_err
             at /home/joshua/.local/lib/rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/result.rs:591
   7: rustwide::prepare::Prepare::fetch_deps
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/prepare.rs:169
   8: rustwide::prepare::Prepare::fetch_deps
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/prepare.rs:166
   9: rustwide::prepare::Prepare::prepare
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/prepare.rs:44
  10: rustwide::build::BuildDirectory::run
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/build.rs:152
  11: rustwide::build::BuildBuilder::run
             at /home/joshua/.local/lib/cargo/registry/src/github.com-1ecc6299db9ec823/rustwide-0.13.0/src/build.rs:89
  12: docs_rs::docbuilder::rustwide_builder::RustwideBuilder::build_package
             at src/docbuilder/rustwide_builder.rs:303
```

**Comment by jyn514** [comment]
I guess another thing we could do is *just* record the name and version? But at that point I think it makes more sense to fix https://github.com/rust-lang/docs.rs/issues/1011 instead, we don't have any more info available than what's in the index.

**Comment by Nemo157** [comment]
Yep, my thoughts on this was definitely to do it after #1011, so that we can also create the `build` object in the database earlier to attach the log to, instead of only creating that after the end of the build process (also necessary for the ideas I had around #787 so we have something to attach all the other build logs to as the process happened).

**Comment by syphar** [comment]
solved with #2467
